packages/next/src/server/app-render/app-render.tsx

```diff
@@ -267,7 +273,9 @@ function makeGetDynamicParamFromSegment(
 
     let value = params[key]
 
-    if (Array.isArray(value)) {
+    if (fallbackRouteParams && fallbackRouteParams.has(segmentParam.param)) {
+      value = fallbackRouteParams.get(segmentParam.param)!
```

Comment:
I think technically this `!` isn't necessary since the `params` value is already typed as `any` (which should probably be fixed 😆 )